### ISSUE TITLE
--sandbox_add_mount_pair ignored on OSX

### ISSUE THREAD

**Issue by greggdonovan** [bazelbuild/bazel#4150]
Please provide the following information. The more we know about your system and use case, the more easily and likely we can help.

### Description of the problem / feature request / question:

I would like to use `--sandbox_add_mount_pair` to mount specific directories into the sandbox filesystem on both OSX and Linux, but [this support was only added on Linux](https://github.com/bazelbuild/bazel/commit/18e6b410a7f4e24b67740978d101cd665edea6cc).

### If possible, provide a minimal example to reproduce the problem:

[DarwinSandboxedSpawnRunner](https://github.com/bazelbuild/bazel/blob/04f2b03d351d3de16601dff2290f40643f2cb16e/src/main/java/com/google/devtools/build/lib/sandbox/DarwinSandboxedSpawnRunner.java) is missing `sandboxAdditionalMounts`, but [LinuxSandboxedSpawnRunner](https://github.com/bazelbuild/bazel/blob/04f2b03d351d3de16601dff2290f40643f2cb16e/src/main/java/com/google/devtools/build/lib/sandbox/LinuxSandboxedSpawnRunner.java) has them.

### Environment info
* Operating System:

macOS Sierra 10.12.6

* Bazel version (output of `bazel info release`):

release 0.7.0-homebrew

**Comment by meteorcloudy** [comment]
//cc @xingao267 @philwo 
Can we provide the same support for OS X?

**Comment by mboes** [comment]
I would likewise be very interested in this feature. It would allow using rules_nixpkgs without requiring any globally installed tool (which can only be installed locally if extra directories are mapped into the sandbox).

**Comment by jin** [comment]
Moving to the local exec team and back into the untriaged pool.

**Comment by jmmv** [comment]
Bind mounts don't exist on macOS (unless you use FUSE, which we don't rely on by default at this point), so I think the only way we could offer this right now is by replicating the tree into the sandbox using symlinks. Would that be sufficient? Are you still interested in this feature?

**Comment by jmmv** [comment]
If the flag exists on macOS, and it does, then we should make it work. Given what I said in the last reply, we cannot make it work as "mounts" (unless we use sandboxfs), but the symlinking approach may be a good compromise.

**Comment by greggdonovan** [comment]
@jmmv Thanks! I'm not actively using macOS, but we most of our Bazel users are. Combined with platform specific bazelrc this would be a nice way to make our builds more hermetic.

**Comment by honnix** [comment]
This feature request seems to be still valid. I don't exactly know how sandbox-exec works, but it seems to be possible to add more here https://github.com/bazelbuild/bazel/blob/04f2b03d351d3de16601dff2290f40643f2cb16e/src/main/java/com/google/devtools/build/lib/sandbox/DarwinSandboxedSpawnRunner.java#L254-L258. Is that right?

**Comment by tjgq** [comment]
Yes, I think this FR still makes sense, although we might not be able to prioritize it. cc @oquenchil

**Comment by honnix** [comment]
Trying this again. Can some one help review the PR? Thanks.

**Comment by oquenchil** [comment]
Commented on the PR.
